src/util/autocomplete.ts

```diff
@@ -0,0 +1,354 @@
+    super(questions, rl, answers)
+    if (!this.opt.searchOpts) {
+      this.throwParamError('searchOpts')
+    }
+
+    this.fuse = new Fuse(this.opt.options, this.opt.searchOpts)
+
+    this.currentChoices = new Choices([], {})
+
+    this.firstRender = true
+
+    // Make sure no default is set (so it won't be printed)
+    this.initialValue = this.opt.default as string
+
+    if (!this.opt.suggestOnly) {
+      this.opt.default = null
+    }
+
+    const shouldLoop = this.opt.loop === undefined ? true : this.opt.loop
+    this.paginator = new Paginator(this.screen, {isInfinite: shouldLoop})
+  }
+
+  public async _run(cb: (state: unknown) => void): Promise<AutocompleteSearch> {
+    this.done = cb
+
+    if (Array.isArray(this.rl.history)) {
+      this.rl.history = []
+    }
+
+    const events = observe(this.rl)
+
+    // eslint-disable-next-line unicorn/consistent-function-scoping
+    const dontHaveAnswer = (): boolean => this.answer === undefined
+
+    void events.line.pipe(takeWhile(dontHaveAnswer)).forEach(this.onSubmit.bind(this))
+    // @ts-ignore
+    void events.keypress.pipe(takeWhile(dontHaveAnswer)).forEach(this.onKeypress.bind(this))
+
+    // Call once at init
+    this.search()
+
+    return Promise.resolve(this)
+  }
+
+  public render(error?: string): void {
+    // Render question
+    let content = this.getQuestion()
+    let bottomContent = ''
+
+    if (this.firstRender) {
+      const suggestText = this.opt.suggestOnly ? ', tab to autocomplete' : ''
+      content += chalk.dim(`(Use arrow keys or type to search ${suggestText})`)
+    }
+
+    // Render choices or answer depending on the state
+    if (this.status === 'answered') {
+      content += chalk.cyan(this.shortAnswer || this.answerName || this.answer)
+    } else if (this.searching) {
+      content += this.rl.line
+      bottomContent += '  ' + chalk.dim(this.opt.searchText || 'Searching...')
+    } else if (this.nbChoices) {
+      const choicesStr = renderList(this.currentChoices, this.selected)
+      content += this.rl.line
+      const indexPosition = this.selected
+      let realIndexPosition = 0
+      this.currentChoices.choices.every((choice, index) => {
+        if (index > indexPosition) {
+          return false
+        }
+
+        if (choice.type === 'separator') {
+          return false
+        }
+
+        const name = choice.name
+        realIndexPosition += name ? name.split('\n').length : 0
+        return true
+      })
+      bottomContent += this.paginator.paginate(choicesStr, realIndexPosition, this.opt.pageSize)
+      if (this.opt.footer) {
+        const footer = this.opt.footer(this.currentChoices, this.selected, bottomContent)
+        if (footer) bottomContent += `\n${new Inquirer.Separator().line}\n${footer}`
+      }
+    } else {
+      content += this.rl.line
+      bottomContent += '  ' + chalk.yellow(this.opt.emptyText || 'No results...')
```

Comment:
Not a huge deal but string templates are a little nicer to read imo
```suggestion
      bottomContent += `  ${chalk.yellow(this.opt.emptyText ?? 'No results...')}`
```